go/proof_data_test.go

```diff
@@ -69,3 +69,135 @@ func CheckAgainstSpecTestData(t *testing.T) map[string]CheckAgainstSpecTestStruc
 	}
 	return cases
 }
+
+type EmptyBranchTestStruct struct {
+	Op     *InnerOp
+	Spec   *InnerSpec
+	IsTrue bool
+	IsLeft bool
+}
+
+var InnerSpecWithEmptyChild = InnerSpec{
+	ChildOrder:      []int32{0, 1},
+	ChildSize:       32,
+	MinPrefixLength: 1,
+	MaxPrefixLength: 1,
+	EmptyChild:      []byte("32_empty_child_placeholder_bytes"),
+	Hash:            HashOp_SHA256,
+}
+
+func EmptyBranchTestData(t *testing.T) []EmptyBranchTestStruct {
+	return []EmptyBranchTestStruct{
+		EmptyBranchTestStruct{
+			Op: &InnerOp{
+				Prefix: []byte{1},
+				Suffix: InnerSpecWithEmptyChild.EmptyChild,
+			},
+			Spec:   &InnerSpecWithEmptyChild,
+			IsTrue: true,
+			IsLeft: false,
+		},
+		EmptyBranchTestStruct{
+			Op: &InnerOp{
+				Prefix: append([]byte{1}, make([]byte, 32)...),
+				Suffix: InnerSpecWithEmptyChild.EmptyChild,
+			},
+			Spec:   &InnerSpecWithEmptyChild,
+			IsTrue: true,
+			IsLeft: false,
+		},
+		EmptyBranchTestStruct{
+			Op: &InnerOp{
+				Prefix: []byte{1},
+				Suffix: make([]byte, 32),
+			},
+			Spec:   &InnerSpecWithEmptyChild,
+			IsTrue: false,
+			IsLeft: false,
+		},
+		EmptyBranchTestStruct{
+			Op: &InnerOp{
+				Prefix: []byte{1},
```

Comment:
again, this is invalid.

maybe add another check if it even matches the spec, but here I would assume we only want to focus on valid ops and just test the left/right value.

Please verify the spec with `op.CheckAgainstSpec(InnerSpecWithEmptyChild)`. And assert that is true for all these cases before the other checks. 

Maybe you can pull out the invalid tests case to another test that just checks validity (and add a valid case or two there?)

Reply:
I tried making a new test for inner op validity, but currently `(InnerOp) CheckAgainstSpec` doesn't check for total byte length. So I added that check, but that causes `TestCheckAgainstSpec` to fail all the "valid" cases, which actually use[ inner ops](https://github.com/confio/ics23/blob/e736db6536ced19ed677b4d4d00ade63d47d4a3d/testdata/TestCheckAgainstSpecData.json#L13) with invalid prefix bytes. I could change those, but it seems out of scope for this PR. I will just remove the invalid cases here.